BLUESPAWN-client/src/util/eventlogs/EventLogs.cpp

```diff
@@ -6,296 +6,296 @@
-		else
-			LOG_ERROR("EventLogs::QueryEvents: EvtQuery failed with " + std::to_string(status));
-
-		goto cleanup;
-	}
+	std::shared_ptr<EVENT_DETECTION> EventLogs::EventLogItemToDetection(const EventLogItem& pItem) {
+		auto detect = std::make_shared<EVENT_DETECTION>(0, 0, L"", L"", L"");
 
-	int numFound = 0;
-	status = ProcessResults(hResults, reaction, &numFound, params);
+		detect->eventID = pItem.GetEventID();
+		detect->channel = pItem.GetChannel();
+		detect->eventRecordID = pItem.GetEventRecordID();
+		detect->timeCreated = pItem.GetTimeCreated();
+		detect->rawXML = pItem.GetXML();
+		detect->params = pItem.GetProperties();
 
-cleanup:
+		return detect;
+	}
+
+	bool IsChannelOpen(const std::wstring& channel) {
+		EVT_HANDLE hChannel = NULL;
+		DWORD status = ERROR_SUCCESS;
+		PEVT_VARIANT pProperty = NULL;  
+		PEVT_VARIANT pTemp = NULL;
+		DWORD dwBufferSize = 0;
+		DWORD dwBufferUsed = 0;
+
+		// Open the channel config
+		hChannel = EvtOpenChannelConfig(NULL, channel.c_str(), 0);
+		if (NULL == hChannel)
+		{
+			LOG_ERROR(L"EventLogs::IsChannelOpen: EvtOpenChannelConfig failed with " + std::to_wstring(GetLastError()) + L" for channel " + channel);
+			return false;
+		}
 
-	if (hResults)
-		EvtClose(hResults);
+		// Attempt to get the channel property
+		if (!EvtGetChannelConfigProperty(hChannel, EvtChannelConfigEnabled, 0, dwBufferSize, pProperty, &dwBufferUsed))
+		{
+			status = GetLastError();
+			if (ERROR_INSUFFICIENT_BUFFER == status) {
+				dwBufferSize = dwBufferUsed;
+				pTemp = (PEVT_VARIANT)realloc(pProperty, dwBufferSize);
+
+				if (pTemp) {
+					pProperty = pTemp;
+					pTemp = NULL;
+					EvtGetChannelConfigProperty(hChannel, EvtChannelConfigEnabled, 0, dwBufferSize, pProperty, &dwBufferUsed);
+				}
+				else {
+					if (pProperty)
+						free(pProperty);
+
+					LOG_ERROR(L"EventLogs::IsChannelOpen: realloc failed for channel " + channel);
+					return false;
+				}
+			}
 
-	if (status == ERROR_SUCCESS)
-		return numFound;
-	return -1;
-}
+			if (ERROR_SUCCESS != (status = GetLastError())) {
+				LOG_ERROR(L"EventLogs::IsChannelOpen: EvtGetChannelConfigProperty failed with " + std::to_wstring(GetLastError()) + L" for channel " + channel);
+				return false;
+			}
 
-std::unique_ptr<EventSubscription> EventLogs::subscribe(LPWSTR pwsPath, unsigned int id, Reactions::HuntTriggerReaction& reaction, DWORD * pStatus) {
-	*pStatus = ERROR_SUCCESS;
-	EVT_HANDLE hSubscription = NULL;
+		}
+		if (pProperty)
+			free(pProperty);
 
-	auto query = std::wstring(L"Event/System[EventID=") + std::to_wstring(id) + std::wstring(L"]");
-	auto wquery = query.c_str();
+		return pProperty->BooleanVal;
+	}
 
-	auto eventSub = std::make_unique<EventSubscription>(reaction);
+	bool OpenChannel(const std::wstring& channel) {
+		EVT_HANDLE hChannel = NULL;
+		DWORD status = ERROR_SUCCESS;
+		EVT_VARIANT ChannelProperty;
+		DWORD dwBufferSize = sizeof(EVT_VARIANT);
+		DWORD dwBufferUsed = 0;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'dwBufferUsed' is assigned a value that is never used.](https://app.codacy.com/manual/ION28/BLUESPAWN/pullRequest?prid=5092780)